File: TypeScript Projects v2/Project 1.ts
// Function to generate Fibonacci series iteratively
function fibonacciIterative(n: number): number[] {
    let series: number[] = [0, 1];
    for (let i = 2; i < n; i++) {
        series.push(series[i - 1] + series[i - 2]);
    }
    return series.slice(0, n);
}

// Function to generate Fibonacci series recursively
function fibonacciRecursive(n: number, series: number[] = [0, 1]): number[] {
    if (series.length >= n) return series.slice(0, n);
    series.push(series[series.length - 1] + series[series.length - 2]);
    return fibonacciRecursive(n, series);
}

// Function to get user input and display Fibonacci series
function generateFibonacci(): void {
    let numStr = prompt("Enter the number of Fibonacci terms: ");
    let num = numStr ? parseInt(numStr) : 0;
    
    if (isNaN(num) || num <= 0) {
        console.log("Please enter a valid positive number.");
        return;
    }
    
    console.log("\nUsing Iterative Approach:");
    console.log(fibonacciIterative(num));
    
    console.log("\nUsing Recursive Approach:");
    console.log(fibonacciRecursive(num));
}

// Run the function
generateFibonacci();

File: Python Projects v2/Project 1.py
import math

def is_prime_iterative(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def is_prime_recursive(n, i=2):
    if n < 2:
        return False
    if i > math.sqrt(n):
        return True
    if n % i == 0:
        return False
    return is_prime_recursive(n, i + 1)

def main():
    try:
        num = int(input("Enter a number to check if it's prime: "))
        
        print("\nUsing Iterative Approach:")
        if is_prime_iterative(num):
            print(f"{num} is a prime number.")
        else:
            print(f"{num} is not a prime number.")
        
        print("\nUsing Recursive Approach:")
        if is_prime_recursive(num):
            print(f"{num} is a prime number.")
        else:
            print(f"{num} is not a prime number.")
    
    except ValueError:
        print("Please enter a valid integer!")

if __name__ == "__main__":
    main()